src/carousel/carousel.ts

```diff
@@ -202,6 +199,9 @@ export class NgbCarousel implements AfterContentChecked,
       this._start$.next();
       this.activeId = selectedSlide.id;
     }
+
+    // we get here after the interval fires or any external API call like next(), prev() or select()
+    this._cd.detectChanges();
```

Comment:
Shouldn't this be `markForCheck`? What would happen in a scenario where click on prev / next is calling prev / next on a carousel _and_ is adding a new slide (ex. from `ngFor`). Would the content of a carousel be checked if we just call detect changes on the carousel itself?

Reply:
> markForCheck would only refresh it on next CD, isn't it correct ?

I would say: in the current CD (the one that would be run at the end of JS loop, before any rendering is done). So I believe that this is the behaviour we actually want.

We should add a test for the above scenario to be sure.